keps/sig-node/127-usernamespaces-support/README.md

```diff
@@ -0,0 +1,1122 @@
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+#### Alpha
+
+- [ ] Support for `Cluster` and `Host` modes implemented.
+- [ ] Support implemented in CRI-O.
+- [ ] Support implemented in containerd.
+- [ ] Unit test coverage.
+- [ ] Support for `Pod` mode discused and implemented.
+
+#### Beta
+
+- [ ] Feedback from alpha is addressed.
+- [ ] E2E test coverage.
+- [ ] There are well-documented use cases of this feature.
+
+#### GA
+
+TDB
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+The container runtime will have to be updated in the nodes to support this feature.
+
+The new `user` field in the `NamespaceOption` will be ignored by an old runtime
+without user namespaces support. The container will be placed in the host user
+namespace. It's a responsibility of the user to guarantee that a runtime
+supporting user namespaces is used when this feature is enabled.
+
+An old version of kubelet without user namespaces support can cause some
```

Comment:
old version of the kubelet with new version of container runtime, I guess? Can we clarify what is the version skew in this case?

Reply:
Clarified here https://github.com/kinvolk/kubernetes-enhancements/pull/3/commits/2392a05f9e67ac4532d2d89762885ebcafe42476.